http/src/request/guild/create_guild.rs

```diff
@@ -182,17 +339,42 @@ impl<'a> CreateGuild<'a> {
 
     /// Set the roles to create with the guild.
     ///
-    /// The maximum number of roles that can be provided is 250.
+    /// The maximum number of roles that can be provided is 250. A `Vec` of [`RoleFragment`]
+    /// structs must be provided. These will be turned into instances of `Role`s that are sent to
+    /// Discord. If specifying `Some` for [`RoleFragment.id`], do not use the id of 1.
     ///
     /// # Errors
     ///
     /// Returns [`CreateGuildError::TooManyRoles`] if the number of roles is
     /// over 250.
     ///
     /// [`CreateGuildError::TooManyRoles`]: enum.CreateGuildError.html#variant.TooManyRoles
-    pub fn roles(mut self, roles: Vec<Role>) -> Result<Self, CreateGuildError> {
-        if roles.len() > 250 {
-            return Err(CreateGuildError::TooManyRoles { roles });
+    /// [`RoleFragment`]: struct.RoleFragment.html
+    /// [`RoleFragment.id`]: struct.RoleFragment.html#structfield.id
+    pub fn roles(mut self, fragments: Vec<RoleFragment>) -> Result<Self, CreateGuildError> {
+        if fragments.len() > 250 {
+            return Err(CreateGuildError::TooManyRoles { roles: fragments });
+        }
+
+        let mut roles: Vec<Role> = fragments
```

Comment:
inference

Reply:
The type is specified to avoid a turbofish on collect